shared/reducers/teams.js

```diff
@@ -0,0 +1,137 @@
+// @flow
+import * as TeamsGen from '../actions/teams-gen'
+import * as Constants from '../constants/teams'
+import * as I from 'immutable'
+import * as Types from '../constants/types/teams'
+
+const initialState: Types.State = Constants.makeState()
+
+const rootReducer = (state: Types.State = initialState, action: TeamsGen.Actions): Types.State => {
+  switch (action.type) {
+    case TeamsGen.resetStore:
+      return initialState
+
+    case TeamsGen.setChannelCreationError:
+      return state.set('channelCreationError', action.payload.error)
+
+    case TeamsGen.setTeamCreationError:
+      return state.set('teamCreationError', action.payload.error)
+
+    case TeamsGen.setTeamCreationPending:
+      return state.set('teamCreationPending', action.payload.pending)
+
+    case TeamsGen.setTeamJoinError:
+      return state.set('teamJoinError', action.payload.error)
+
+    case TeamsGen.setTeamJoinSuccess:
+      return state.withMutations(s => {
+        s.set('teamJoinSuccess', action.payload.success)
+        s.set('teamJoinSuccessTeamName', action.payload.teamname)
+      })
+
+    case TeamsGen.setTeamStoreRetentionPolicy:
```

Comment:
already an existing `setTeamRetentionPolicy` action. better name suggestions welcome! (as for all the other new actions)